docs/dev-guide.md

```diff
@@ -255,3 +255,311 @@ which will properly configure and run all microservices.
 Please assure that MQTT microservice has `node_modules` installed, as explained in _MQTT Microservice_ chapter.
 
 > N.B. `make rundev` actually calls helper script `scripts/run.sh`, so you can inspect this script for the details.
+
+## Events
+In order to be easily integratable system, Mainflux is using [Redis Streams](https://redis.io/topics/streams-intro) 
+as an event log for event sourcing. Services that are publishing events to Redis Streams 
+are `things` service, and `bootstrap` service, and `mqtt` adapter.
+
+### Things Service
+For every operation that has side effects (that is changing service state) `things` 
+service will generate new event and publish it to Redis Stream called `mainflux.things`. 
+Every event has its own event ID that is automatically generated and `operation` 
+field that can have one of the following values:
+- `thing.create` for thing creation,
+- `thing.update` for thing update,
+- `thing.remove` for thing removal,
+- `thing.connect` for connecting a thing to a channel,
+- `thing.disconnect` for disconnecting thing from a channel,
+- `channel.create` for channel creation,
+- `channel.update` for channel update,
+- `channel.remove` for channel removal.
+
+By fetching and processing these events you can reconstruct `things` service state. 
+If you store some of your custom data in `metadata` field, this is the perfect 
+way to fetch it and process it. If you want to integrate through 
+[docker-compose.yml](https://github.com/mainflux/mainflux/blob/master/docker/docker-compose.yml)
+you can use `mainflux-things-redis` service. Just connect to it and consume events 
+from Redis Stream named `mainflux.things`.
+
+#### Thing create event
+
+Whenever thing is created, `things` service will generate new create event. This 
+event will have the following format:
+```
+1) "1555334740911-0"
+2)  1) "type"
+    2) "device"
+    3) "operation"
+    4) "thing.create"
+    5) "name"
+    6) "d0"
+    7) "id"
+    8) "3c36273a-94ea-4802-84d6-a51de140112e"
+    9) "owner"
+   10) "john.doe@email.com"
+   11) "metadata"
+   12) "{}"
+```
+
+As you can see from this example, every odd field represents field name while every 
+even field represents field value. This is standard event format for Redis Streams.
```

Comment:
event

Reply:
This is not typo, even (opposite of odd).

Reply:
Then, I don't understand your sentence

Reply:
@manuio In Redis, data is not structured in the `key:value` format, but simply in the form of list. Each odd element represents a field name (key), and following even element represents the value of that field, so:
```
...
1) owner
2) john.doe@email
...
```
means that the value of the "owner" key is "john.doe@email.com".